gateway-ha/src/main/java/io/trino/gateway/ha/router/GatewayCookie.java

```diff
@@ -177,10 +177,6 @@ public static GatewayCookie fromCookie(Cookie cookie)
 
     public boolean matchesRoutingPath(String path)
     {
-        if (matchesDeletePath(path)) {
```

Comment:
Do you know why this was there in the first place then?

Reply:
I believe that this logic was there because the original model of the flow was basically 
* identify the first request in a "session" and create a cookie to track it
* if the client makes another request that initiates a session, delete the old cookie and create a new session

This logic was carried over from that model, which doesn't properly handle logout.